### ISSUE TITLE
NGINX plugin does not copy cert to /keys for TCP stream proxy

### ISSUE THREAD

**Issue by henningkessler** [opnsense/plugins#2189]
**Describe the bug**
I had a forum post about this [here](https://forum.opnsense.org/index.php?topic=20850.0)
I am trying to configure a TCP stream proxy for LDAP with the NGINX plugin using a cert from the lets encrypt plugin. Unfortunately starting Nginx always fails with the following error:
```
cannot load certificate "/usr/local/etc/nginx/key/f5e949f2-0d6b-42a8-8c52-9706945f9454.pem": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/usr/local/etc/nginx/key/f5e949f2-0d6b-42a8-8c52-9706945f9454.pem','r') error:2006D080:BIO routines:BIO_new_file:no such file)
```
If check the mentioned directory it is empty or the cert has a length of 0 bytes.

This is the config:
```
...
stream {
...

    # UPSTREAM SERVERS
    upstream upstream123dd4ddf4c74e86bce785b2817d5096 {
        hash $remote_addr consistent;
        server IP:636 weight=1 max_conns=5 max_fails=2 fail_timeout=20;
        server IP:636 weight=1 max_conns=5 max_fails=2 fail_timeout=20;
    }

    # upstream maps


    include opnsense_stream_vhost_plugins/*.conf;

    # servers
    server {
        listen  63636 ssl;
        listen  [::]:63636 ssl;

        access_log  /var/log/nginx/stream_f5e949f2-0d6b-42a8-8c52-9706945f9454.access.log main;
        error_log  /var/log/nginx/stream_f5e949f2-0d6b-42a8-8c52-9706945f9454.error.log info;

        ssl_client_certificate /usr/local/etc/nginx/key/f5e949f2-0d6b-42a8-8c52-9706945f9454_ca.pem;
        ssl_verify_client off;
        ssl_certificate_key /usr/local/etc/nginx/key/f5e949f2-0d6b-42a8-8c52-9706945f9454.key;
        ssl_certificate /usr/local/etc/nginx/key/f5e949f2-0d6b-42a8-8c52-9706945f9454.pem;
        ssl_protocols TLSv1.2 TLSv1.3;
        ssl_dhparam /usr/local/etc/dh-parameters.4096;
        ssl_ciphers 'ECDHE-ECDSA-CAMELLIA256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-CAMELLIA256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-ECDSA-CAMELLIA128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-CAMELLIA128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-CAMELLIA256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-CAMELLIA256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-ECDSA-CAMELLIA128-SHA256:ECDHE-RSA-AES128-SHA256';
        ssl_session_timeout 1d;
        ssl_session_cache shared:sslcachef5e949f20d6b42a88c529706945f9454:50m;
        ssl_session_tickets off;
        ssl_prefer_server_ciphers on;

        proxy_ssl on;
        proxy_pass upstream123dd4ddf4c74e86bce785b2817d5096;
        proxy_protocol off;

    }
}
...
```
After adding a second stream server the certs for the first where copied to /key but the cert for the upstream server still has 0 bytes...

**Expected behavior**
Should work without adding a second stream server? 

**Environment**
OPNsense 20.7.7_1-amd64
Intel Xeon D-1518

**Comment by kulikov-a** [comment]
hi all!
as I mentioned on the forum, it looks like the setup.php does not quite correctly checks the servers arrays on strings 79-83 and 124-128
@henningkessler 

> but the cert for the upstream server still has 0
Based on information from the forum (https://forum.opnsense.org/index.php?topic=20850.msg97260#msg97260), you have not defined a certificate and TLS is not enabled on the upstream. try to enable and select a certificate. and apply the changes

**Comment by henningkessler** [comment]
Unfortunately still the same...

**Comment by folfy** [comment]
If your issue is with the upstream server certificate, this probably relates to the bug fixed in PR #2198.

**Comment by folfy** [comment]
Fix has been merged into master, guess this can be closed once a new version is released.
